mrtt-ui/src/components/CheckboxGroupMangroveWithLabel.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { FormQuestionDiv } from '../styles/forms'
+import { FormLabel } from '@mui/material'
+import { Controller } from 'react-hook-form'
+import CheckboxGroup, { separateOtherValueFromSelectedValues } from './CheckboxGroup/CheckboxGroup'
+import separateOutliersFromInliers from '../library/separateOutliersFromInliers'
+import getOptionsObjectFromQuestionAnswers from '../library/generateOptionsObjectFromQuestionAnswers'
+
+const CheckboxGroupMangroveWithLabel = ({
+  fieldName,
+  options,
+  question,
+  reactHookFormInstance,
+  shouldAddOtherOptionWithClarification
+}) => {
+  const { control: formControl, setValue: setFormValue } = reactHookFormInstance
+  const labelId = `${fieldName}-label`
+  const optionsValueLabels = getOptionsObjectFromQuestionAnswers(options)
```

Comment:
'options' as a list of strings vs 'options' as a list of {label: string, value: string} for the ui. Better naming to come.